### PR TITLE
 Fadi week1 and 2 homework

### PR DESCRIPTION
**Pull request by fadikurdi**

### CODE REVIEW

File: Week1/homework/world-queries.txt [hunk 1]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
 | Alkmaar           |
 +-------------------+
-28 rows in set (0.00 sec)
\ No newline at end of file
+28 rows in set (0.00 sec)
+
+
+2.What is the population of Rotterdam ?
+
+ SELECT name, population FROM city WHERE Name='Rotterdam';
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: :+1:

File: Week1/homework/world-queries.txt [hunk 2]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
++-----------+------------+
+| name      | population |
++-----------+------------+
+| Rotterdam |     593321 |
++-----------+------------+
+1 row in set (0.00 sec)
+
+3.What's the name of all the countries on the continent ‘North America’ ?
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: :+1:

File: Week1/homework/world-queries.txt [hunk 3]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
+| United States                    |
+| Saint Vincent and the Grenadines |
+| Virgin Islands, British          |
+| Virgin Islands, U.S.             |
++----------------------------------+
+37 rows in set (0.00 sec)
+
+4.What's the top 10 countries by SurfaceArea ?
+
+ SELECT Name, SurfaceArea FROM country ORDER BY SurfaceArea DESC LIMIT 10;
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: :+1:

File: Week1/homework/world-queries.txt [hunk 4]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
+| Australia          |  7741220.00 |
+| India              |  3287263.00 |
+| Argentina          |  2780400.00 |
+| Kazakstan          |  2724900.00 |
++--------------------+-------------+
+10 rows in set (0.00 sec)
+
+5.What's the top 10 most populated cities?
+
+ SELECT Name, Population FROM country ORDER BY Population DESC LIMIT 10;
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: Question asked for cities, not countries, but you were pretty close.

File: Week1/homework/world-queries.txt [hunk 5]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
+| Russian Federation |  146934000 |
+| Bangladesh         |  129155000 |
+| Japan              |  126714000 |
+| Nigeria            |  111506000 |
++--------------------+------------+
+10 rows in set (0.00 sec)
+
+6.Select the top 3 countries by population that their names start with the letter ‘P’?
+
+ SELECT Name FROM country WHERE Name LIKE 'P%' ORDER BY Population DESC LIMIT 3;
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: :+1:

File: Week1/homework/world-queries.txt [hunk 6]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
+|   22256000 |
+|   24318000 |
+|   79832000 |
+|   18112000 |
++------------+
+51 rows in set (0.00 sec)
+
+7.1 And what is the population of Asia ?
+
+ SELECT SUM(Population) FROM country WHERE Continent = "Asia" ;
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: Yeah, this is what the question was really asking.

Comment (jasongwartz) [reply Week1/homework/world-queries.txt]: Did you and kemirdin work together on this?

File: Week1/homework/world-queries.txt [hunk 7]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
++-----------------+
+| SUM(Population) |
++-----------------+
+|      3705025700 |
++-----------------+
+1 row in set (0.00 sec)
+
+8.Which languages are spoken in the Region of ‘South America’?
+
+ SELECT language FROM countrylanguage INNER JOIN country ON country.code=countrylanguage.countrycode WHERE country.continent='South America';
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: 👍

File: Week1/homework/world-queries.txt [hunk 8]
```diff
@@ -37,4 +37,249 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
...
+| Goajiro          |
+| Spanish          |
+| Warrau           |
++------------------+
+42 rows in set (0.00 sec)
+
+9.What are the languages spoken on all cities named ‘Barcelona’ (you may need to join 3 tables =) ) ?
+
+
+ SELECT city.Name, language FROM countrylanguage JOIN city ON city.Countrycode = countrylanguage.Countrycode JOIN country ON city.Countrycode = country.Code WHERE city.Name = 'Barcelona';
```
Comment (jasongwartz) [inline Week1/homework/world-queries.txt]: 👍

File: Week1/homework/world-queries.txt (hunk 4 above)
Comment (jamsrahimi) [reply Week1/homework/world-queries.txt]: Hi @fadikurdi or @zydan1, did you fix or comment here yet?

File: Week1/homework/world-queries.txt (hunk 6 above)
Comment (fadikurdi) [reply Week1/homework/world-queries.txt]: no Mr.Jason i did not

File: Week2/todo_app/program.js [hunk 9]
```diff
@@ -1,72 +1,203 @@
...
             callback(null, results);
         });
     }
 
-    create(description, callback) {
-        // Write code and query to create a new TODO item
+    create(description, user_id, callback) {
+        const insertItem =
+            `INSERT INTO todo_items(text,is_completed,user_id)
+            VALUES(${connection.escape(description)} ,0 ,${connection.escape(user_id)} );
```
Comment (jasongwartz) [inline Week2/todo_app/program.js]: Nice job using escaping here.

:+1:

File: Week2/todo_app/program.js [hunk 10]
```diff
@@ -1,72 +1,203 @@
...
+            callback(null, results);
+        });
     }
 
     update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+        const updateItem =
+            `
+            UPDATE todo_items SET text = ${connection.escape(description)}
+              WHERE id = ${connection.escape(id)}
```
Comment (jasongwartz) [inline Week2/todo_app/program.js]: :+1:

File: Week2/todo_app/program.js [hunk 11]
```diff
@@ -1,72 +1,203 @@
...
+            callback(null, results);
+        });
     }
 
     delete(id, callback) {
-        // Write code and query to delete an existing TODO item
+        const deleteItem =
+            `
+            DELETE FROM todo_items 
+            WHERE id = ${connection.escape(id)}
```
Comment (jasongwartz) [inline Week2/todo_app/program.js]: :+1:

File: Week2/todo_app/program.js [hunk 12]
```diff
@@ -1,72 +1,203 @@
...
+            callback(null, results);
+        });
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
+        const tagItem =
+            `
+            INSERT INTO todo_item_tag(todo_item_id,tag_id)
+            VALUES (${connection.escape(todoItemId)} , ${connection.escape(tagId)});
```
Comment (jasongwartz) [inline Week2/todo_app/program.js]: Same comment as for everybody else - this assumes that the tag already exists in the database, like if it has been used before. How do you handle a case where the tag doesn't already exist?

File: Week2/todo_app/program.js [hunk 13]
```diff
@@ -1,72 +1,203 @@
...
-        
-    untagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query remove a tag from a TODO item
+
+    unTagTodoItem(todoItemId, tagId, callback) {
+        const unTagItem =
+            `
+            DELETE FROM todo_item_tag
+            WHERE todo_item_id= ${todoItemId}
+            AND tag_Id = ${connection.escape(tagId)}
```
Comment (jasongwartz) [inline Week2/todo_app/program.js]: :+1:

File: Week2/todo_app/program.js [hunk 14]
```diff
@@ -1,72 +1,203 @@
...
+            }
+            callback(null, results);
+        });
     }
 
     markCompleted(todoItemId, callback) {
-        // Write code to mark a TODO item as completed
+        const markItem =
+            `UPDATE todo_items SET is_completed = 1
+             WHERE id=${connection.escape(todoItemId)}
```
Comment (jasongwartz) [inline Week2/todo_app/program.js]: :+1:

**Pull request closed without merging**